guides/plugins/apps/local-development/app-development-with-docker.md

```diff
@@ -0,0 +1,175 @@
+# App development with docker
+
+This guide will walk you through the process of combining Shopware and your app in one setup.
+
+## File Structure
+
+At first, you should clone the app template from [GitHub](https://github.com/shopwareLabs/AppTemplate) and create a `manifest.xml` for your app. Or for a fully working example based on this template watch out for our [appExample](https://github.com/shopwareLabs/AppExample).  
+For further information about the `manifest.xml` have a look at our [documentation](https://docs.shopware.com/en/shopware-platform-dev-en/app-system-guide/setup#manifest-file).  
+To easily create a `manifest.xml` take a look at our [create-manifest command](../platform-sh-template.md#create-the-manifestxml).
+
+Your file structure should look as follows:
+```
+...
+│
+├──development
+│  ├──custom
+│  │  └───apps
+│  │      └───yourAppName
+│  │          └───manifest.xml
+│  │
+│  ├──platform
+│  └──...
+│
+└──shopwareAppTemplate
+...
+```
+
+## Combining both in one docker setup
+
+Once your Shopware development setup is ready to go you need to add your app to it. This is done by adding the services to your `development/docker-compose.yml`.  
+At first, you need to add two networks. One for your app system and another one for combining the app system with Shopware.
+  
+This is done by simply adding the networks `appSystem` and `development` to your existing ones:
+```Yaml
+networks:
+    shopware:
+    appSystem:
+    development:
+```
+The `appSystem`-network is only for your app server and the app database.  
+The `development`-network is used to combine your app server with the Shopware server.  
+
+### Adding the app server
+
+Now you need to define the `services` in your `development/docker-compose.yml`. Insert the following to firstly add your app server.   
+```Yaml
+services:
+[...]
+  example_app_server:
+    image: shopware/development:local
+    volumes:
+      - "../shopwareAppTemplate:/app"
+      - "~/.composer:/.composer"
+    environment:
+      CONTAINER_UID: 1000
+      APPLICATION_UID: 1000
+      APPLICATION_GID: 1000
+    ports:
+      - "127.0.0.1:7777:8000"
+    networks:
+      appSystem:
+      development:
+        aliases:
+          - example
+```
+
+This adds a new container to your docker setup running your app server's code. The new container is available at the networks `appSystem` and `development`.  
+In the `development`-network your app server has the alias `example`. This will be the url which your Shopware server needs to communicate with. This is also the url which you should use in your `manifest.xml` except for iframes.  
+The `volumes` represents the relative path to your app.  
+And `ports` exposes port `8000` to `127.0.0.1:7777` to us so that we can go to `127.0.0.1:7777` or `localhost:7777` to directly connect to your app server. This will come in handy when we register your own modules to use iframes.  
```

Comment:
```suggestion
And `ports` exposes port `8000` to `127.0.0.1:7777` to us so that we can visit `127.0.0.1:7777` or `localhost:7777` to directly connect to your app server. This will come in handy when we register our own modules to use iframes.  
```